File: PickerControls/FileEntry.cs
using System;

namespace PickerControls
{
    public sealed class FileEntry
    {
        public string Path { get; }
        public string Label { get; }
        public long Size { get; }
        public DirectoryEntry Parent { get; }
        public FileEntry(string path, string label, long size, DirectoryEntry parent)
        {
            if (path.EndsWith(@"\"))
            {
                throw new ArgumentException("Must be a filename", nameof(path));
            }

            Path = path;
            Label = label;
            Size = size;
            Parent = parent;
        }
    }
}

File: PickerDesktopHosting/Control/FileEntry.cpp
#include "pch.h"
#include "FileEntry.h"
#if __has_include("FileEntry.g.cpp")
#include "FileEntry.g.cpp"
#endif

using namespace winrt;

namespace winrt::Control::implementation
{
    FileEntry::FileEntry()
    {
    }

    FileEntry::FileEntry(DirectoryEntry parent, hstring path, hstring label, int64_t size) : m_Parent(parent), m_Path(path), m_Label(label), m_Size(size)
    {
    }

    DirectoryEntry FileEntry::Parent()
    {
        return m_Parent;
    }

    winrt::hstring FileEntry::Path()
    {
        return m_Path;
    }

    winrt::hstring FileEntry::Label()
    {
        return m_Label;
    }

    int64_t FileEntry::Size()
    {
        return m_Size;
    }
}

